utils/shared.py

```diff
@@ -0,0 +1,12 @@
+import numpy as  np
+
+
+def get_l2norm(A, n_iter=100):
+    if isinstance(A, np.ndarray):
+        return np.linalg.norm(A, ord=2)
+    else:
+        x = np.random.randn(A.shape[1])
```

Comment:
The way the operator is computed `A.T @ A` or `A @ A.T` should be decided depending on the shape of `A` no?

Reply:
The factorized version is 10x faster than the non-factorized one:
![image](https://user-images.githubusercontent.com/3321081/171841871-cda7cb06-fda7-417e-a812-a118925a9601.png)